packages/lesswrong/lib/collections/ultraFeedEvents/newSchema.ts

```diff
@@ -0,0 +1,135 @@
+import { DEFAULT_CREATED_AT_FIELD, DEFAULT_ID_FIELD } from "@/lib/collections/helpers/sharedFieldConstants";
+
+const ALLOWED_COLLECTION_NAMES = ["Posts", "Comments", "Spotlights"];
+const ALLOWED_EVENT_TYPES = ["served", "viewed", "expanded"];
+
+const schema = {
+  _id: DEFAULT_ID_FIELD,
+  createdAt: DEFAULT_CREATED_AT_FIELD,
+
+  documentId: {
+    database: {
+      type: "TEXT",
+      nullable: false,
+    },
+    graphql: {
+      outputType: "String",
+      inputType: "String!",
+      canRead: ["admins"],
+      canCreate: ["members"],
+      validation: {
+        optional: false,
```

Comment:
You don't need to manually specify `optional: false` in the validation section; that's the default.  You only need to specify `optional: true` for fields which might get inserted and are not, in fact, required to be provided by the user when creating a new record.